testsuite/tests/tool-toplevel/redefinition_hints.compilers.reference

```diff
@@ -0,0 +1,40 @@
+module Empty : sig  end
+type u = A
+type v = B
+module type S = sig  end
+val m : (module S) = <module>
+module M : sig type 'a t = X of 'a end
+val x : (u * v * (module S)) M.t = M.X (A, B, <module>)
+module type S = sig  end
+val m : (module S) = <module>
+type u = A
+type v = B
+module M : sig type 'a t = X of 'a end
+val y : (u * v * (module S)) M.t = M.X (A, B, <module>)
+Line 2, characters 4-5:
+2 | x = y;;
+        ^
+Error: This expression has type (u/1 * v/1 * (module S/1)) M/1.t
+       but an expression was expected of type
+         (u/2 * v/2 * (module S/2)) M/2.t
+       Hint: The types v and u have been defined multiple times in this
+         toplevel session. Some toplevel values still refer to old versions
+         of those types. Did you try to redefine them?
+       Hint: The module M has been defined multiple times in this toplevel
+         session. Some toplevel values still refer to old versions of this
+         module. Did you try to redefine them?
+       Hint: The module type S has been defined multiple times in this
+         toplevel session. Some toplevel values still refer to old versions
+         of this module type. Did you try to redefine them?
```

Comment:
We could certainly group the plural and singular version, I was being unnecessarily fancy.

I feel that grouping all "namespace" together is a bit more problematic because for instance `s` is a valid name for both module type and a type. Similarly, `S` is valid for both `signature` and modules`. Thus I feel that it might help to precise the namespace.

Moreover, outside of hand-written examples, it seems to be quite improbable to have an error on multiple categories.

Reply:
This could read `the types u and v, the module M and the module type S ha(s)(ve) been defined...`.

Reply:
This is another option indeed.

Reply:
But this means a list printer with two separators (here `,` and `and`).

Reply:
(In the end we decided to stay with the current state.)